### PR TITLE
NFT-942: Fix incorrect message on VaultWriteButton

### PR DESCRIPTION
**Pull request by adamgobes**

### CODE REVIEW

File: components/Controllers/OpenVault/VaultWriteButton.tsx [hunk 1]
```diff
@@ -143,8 +145,27 @@ export function VaultWriteButton({
 
   const buttonText = useMemo(() => {
     if (!oracleInfoSynced) return 'Waiting for oracle...';
-    return vaultHasDebt ? 'Update Loan' : 'Borrow';
-  }, [oracleInfoSynced, vaultHasDebt]);
+    const updateOrBorrowText =
+      vaultHasDebt || vaultCollateralCount > 0 ? 'Update Loan' : 'Borrow';
+    if (amount.isZero()) {
+      if (depositNFTs.length > 0) {
```
Comment (wilsoncusack) [inline components/Controllers/OpenVault/VaultWriteButton.tsx:151]: if they are depositing AND withdrawing we should say "Update Loan"

File: components/Controllers/OpenVault/VaultWriteButton.tsx [hunk 2]
```diff
@@ -143,8 +145,27 @@ export function VaultWriteButton({
...
+        return 'Add NFT';
```
Comment (wilsoncusack) [inline components/Controllers/OpenVault/VaultWriteButton.tsx:152]: Might be multiple?

File: components/Controllers/OpenVault/VaultWriteButton.tsx [hunk 3]
```diff
@@ -143,8 +145,27 @@ export function VaultWriteButton({
...
+      } else if (withdrawNFTs.length > 0) {
+        return 'Withdraw NFT';
```
Comment (wilsoncusack) [inline components/Controllers/OpenVault/VaultWriteButton.tsx:154]: might be multiple?

File: components/Controllers/OpenVault/VaultWriteButton.tsx (hunk 1 above)
Comment (adamgobes) [reply components/Controllers/OpenVault/VaultWriteButton.tsx:151]: yeah you're right

File: components/Controllers/OpenVault/VaultWriteButton.tsx [hunk 4]
```diff
@@ -143,8 +145,27 @@ export function VaultWriteButton({
 
   const buttonText = useMemo(() => {
     if (!oracleInfoSynced) return 'Waiting for oracle...';
-    return vaultHasDebt ? 'Update Loan' : 'Borrow';
-  }, [oracleInfoSynced, vaultHasDebt]);
+    const updateOrBorrowText =
+      vaultHasDebt || vaultCollateralCount > 0 ? 'Update Loan' : 'Borrow';
+    if (amount.isZero()) {
+      if (depositNFTs.length > 0 && withdrawNFTs.length === 0) {
+        return 'Add NFT(s)';
```
Comment (wilsoncusack) [inline components/Controllers/OpenVault/VaultWriteButton.tsx:152]: we should have a conditional on this "s", easy enough and "(s)" looks janky

Comment (adamgobes) [reply components/Controllers/OpenVault/VaultWriteButton.tsx]: good call

File: components/Controllers/OpenVault/VaultWriteButton.tsx [hunk 5]
```diff
@@ -143,8 +145,27 @@ export function VaultWriteButton({
 
   const buttonText = useMemo(() => {
     if (!oracleInfoSynced) return 'Waiting for oracle...';
-    return vaultHasDebt ? 'Update Loan' : 'Borrow';
-  }, [oracleInfoSynced, vaultHasDebt]);
+    const updateOrBorrowText =
```
Comment (wilsoncusack) [inline components/Controllers/OpenVault/VaultWriteButton.tsx:148]: think we can maybe cleanup
```suggestion
    if (depositNFT.length == 0 && withdrawNFTs.length> 0) {
     "withdraw"
     } else if (depositNFT.length > 0 && withdrawNFTs.length == 0 && amount.isZero()) {
     "deposit"
     } else if (depositNFT.length > 0 && withdrawNFTs.length == 0 && !amount.isZero()) {
     "Borrow"
     } else {
     "Update Loan"
     }
```

Comment (adamgobes) [reply components/Controllers/OpenVault/VaultWriteButton.tsx:148]: need amount.isZero() on the first if statement though? lots of redundant amount checks imo

**Review by wilsoncusack — APPROVED** [review]

File: components/Controllers/OpenVault/VaultWriteButton.tsx (hunk 5 above)
Comment (adamgobes) [reply components/Controllers/OpenVault/VaultWriteButton.tsx:148]: think its better to separate by whether amount is zero or not

**Pull request merged**